webapp/channels/src/components/user_settings/plugin/index.tsx

```diff
@@ -45,17 +45,29 @@ const PluginTab = ({
                 <SettingDesktopHeader text={headerText}/>
                 <PluginAction action={settings.action}/>
                 <div className='divider-dark first'/>
-                {settings.sections.map(
-                    (v) =>
-                        (<React.Fragment key={v.title}>
+                {settings.sections.map((v) => {
+                    let sectionEl;
+                    if ('component' in v) {
+                        const CustomComponent = v.component;
+                        sectionEl = (<CustomComponent/>);
```

Comment:
We don't usually do this, but I wonder if we should start adding error boundaries around these "arbitrary rendered components". 0/5.

Reply:
@larkox I like the idea. What would that look like in practice?

Reply:
I thought there was something "out of the box" in react, but not really.

But this library would do the trick: https://github.com/bvaughn/react-error-boundary

Since these are always inside lists, you probably would have to create a new component that will just do something like:

```
const BoundedSectionComponent({component}) {
    const { showBoundary } = useErrorBoundary();
    if (showBoundary) {
        return (<div>Error loading plugin</div>);
    }
    return (<Component/>);
}
```

Reply:
Thanks, @larkox; appreciate the hint. Will dig further and see if we can make it work.

Reply:
@streamer45 Sorry for the back and forth. We have something that should be working already in `webapp/channels/src/plugins/pluggable/error_boundary.tsx`. You could take a look at that too, and maybe reuse it. 0/5

Reply:
Perfect, will look at that first to see if we can avoid the dependency :+1:

Reply:
Added it in https://github.com/mattermost/mattermost/pull/27535/commits/d47804fb650624e82978149c3ec0261c74541fc4. I tested it a bit. It's not exactly bulletproof, but it should cover most rendering issues.